contracts/p1/BackingManager.sol

```diff
@@ -149,22 +153,26 @@ contract BackingManagerP1 is TradingP1, IBackingManager {
          * rToken.basketsNeeded to the current basket holdings. Haircut time.
          */
 
+        (TradingContext memory ctx, Registry memory reg) = tradingContext(basketsHeld);
         (
             bool doTrade,
             TradeRequest memory req,
             TradePrices memory prices
-        ) = RecollateralizationLibP1.prepareRecollateralizationTrade(this, basketsHeld);
+        ) = RecollateralizationLibP1.prepareRecollateralizationTrade(ctx, reg);
 
         if (doTrade) {
+            IERC20 sellERC20 = req.sell.erc20();
+
             // Seize RSR if needed
-            if (req.sell.erc20() == rsr) {
+            if (sellERC20 == rsr) {
                 uint256 bal = req.sell.erc20().balanceOf(address(this));
                 if (req.sellAmount > bal) stRSR.seizeRSR(req.sellAmount - bal);
```

Comment:
I think it's fine? Agree that `tokensOut[rsr]` contains balances of RSR from both the BackingManager and StRSR addresses. There are no balances out on trade because `tradesOpen` is 0.